IntelFsp2Pkg/FspSecCore/SecFsp.c

```diff
@@ -88,8 +88,9 @@ SecGetPlatformData (
       if (*(StackPtr - 1) == FSP_MCUD_SIGNATURE) {
         //
         // This following data was pushed onto stack after TempRamInit API
+        // 64-bit FSP-T pushes these values as QWORDs
         //
-        DwordSize = 4;
+        DwordSize = (4*sizeof (UINTN))/sizeof (UINT32);

```

Comment:
how about sizeof (UINTN)

DwordSize = sizeof (UINTN);

Reply:
do we need to check in line 101 as well?

Reply:
Sorry, I didn't go through details, why this PR is Open still?

Reply:
@AshrafAliS sorry for the delay. I haven't been following this thread.
The DwordSize variable name is a little misleading. It should maybe be DwordCount instead since it indicates the number of dwords to be copied/skipped on the stack. To be more clear, this whole section could be rewritten to not assume the size of the words in the variable names. in ia32 there are 4 DWORDs on the stack to be copied here: MicrocodeRegionBase, MicrocodeRegionSize, CodeRegionBase, CodeRegionSize. In x64 these are 4 QWORDs. 

There is no change to the performance data checked in line 101 in x64 since the TSC data size is the same, so no need to change that.